src/annotation/index.ts

```diff
@@ -6,6 +6,25 @@ import {
   AuditAction,
 } from "./interfaces";
 
+/* eslint-disable */
```

Comment:
You can, but there were like four different `any`-related lint errors that occur, plus "Assignment to property of function parameter 'descriptor'", which seems to just forbid method decorators entirely because AFAIK you have to assign to `descriptor.value` to do this.

Reply:
>which seems to just forbid method decorators entirely because AFAIK you have to assign to descriptor.value to do this.

yea, this is definitely a valid place to ignore that rule

Reply:
add

```json
        "no-param-reassign": [
          "error",
          {
            props: true,
            ignorePropertyModificationsFor: ["descriptor"], // Decorators
          },
        ],
```

to the es=lint file and we can remove the inline comments (especially if we are going to use more decorators)

Reply:
Nevermind though, Craig's managed to fix the lint errors without ignoring anything in the end (except for "Assignment to property of function parameter 'descriptor'" (aka `no-param-reassign`), which we configured `.eslintrc.js` to ignore just for descriptors, in case we ever want to add more decorators).